atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java

```diff
@@ -184,6 +192,10 @@ protected boolean shouldStopRetrying(int numTimesFailed) {
                 timelockService.startIdentifiedAtlasDbTransactionBatch(conditions.size());
         Preconditions.checkState(conditions.size() == responses.size(), "Different number of responses and conditions");
         try {
+            lockWatchValueCache.processStartTransactions(responses.stream()
```

Comment:
again, this should probably live in the same place as `LockWatchEventCache` and its processing.

Reply:
there may even be an argument of pairing the two of them in some handy interface.

Reply:
Yeah, there is a comment, this should live in transaction starter, but I need to do more annoying wiring first

Reply:
Did the wiring, will look into unifying the caches to make code slightly nicer